core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/udf/UserDFOperator.java

```diff
@@ -0,0 +1,322 @@
+package edu.uci.ics.texera.dataflow.udf;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.lang.management.ManagementFactory;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.uci.ics.texera.api.constants.ErrorMessages;
+import edu.uci.ics.texera.api.constants.DataConstants.TexeraProject;
+import edu.uci.ics.texera.api.dataflow.IOperator;
+import edu.uci.ics.texera.api.dataflow.ISourceOperator;
+import edu.uci.ics.texera.api.exception.DataflowException;
+import edu.uci.ics.texera.api.exception.TexeraException;
+import edu.uci.ics.texera.api.schema.AttributeType;
+import edu.uci.ics.texera.api.schema.Schema;
+import edu.uci.ics.texera.api.tuple.Tuple;
+import edu.uci.ics.texera.api.utils.Utils;
+import edu.uci.ics.texera.dataflow.common.AbstractSingleInputOperator;
+import sun.misc.Signal;
+import sun.misc.SignalHandler;
+
+/**
+ * @author Qinhua Huang
+ */
+
+public class UserDFOperator extends AbstractSingleInputOperator implements SignalHandler, ISourceOperator{
+    private UserDFOperatorPredicate predicate;
+    private Schema outputSchema;
+    
+    /* 2 input signal:
+     *      TAG_NULL= "": result NULL
+     *      TAG_LEN = length of text
+     * 3 ouput signal:
+     *      TAG_NULL= "": result NULL
+     *      TAG_WAIT= Character.unsigned:   need to wait for next
+     *      TAG_LEN = length of text
+     */
+    // Signal write to buffer
+    int POSITION_PID = 0;
+    int POSITION_TAG = 10;
+    int POSITION_JSON = 20;
```

Comment:
These properties and other properties below are common to all UDF operators. Please put them inside the predicate definition. While forming the predicate they should be initialised, and the operator formed on that predicate will have access to all properties